types.go

```diff
@@ -258,6 +260,109 @@ type VirtualMachineConfig struct {
 	Boot    string
 	VMGenID string
 	Name    string
+
+	Ides map[string]string
+	Ide0 string
+	Ide1 string
+	Ide3 string
+	Ide4 string
+	Ide5 string
+	Ide6 string
+	Ide7 string
+	Ide8 string
+	Ide9 string
+
+	Scsis map[string]string
+	Scsi0 string
+	Scsi1 string
+	Scsi2 string
+	Scsi3 string
+	Scsi4 string
+	Scsi5 string
+	Scsi6 string
+	Scsi7 string
+	Scsi8 string
+	Scsi9 string
+
+	Satas map[string]string
+	Sata0 string
+	Sata1 string
+	Sata2 string
+	Sata3 string
+	Sata4 string
+	Sata5 string
+	Sata6 string
+	Sata7 string
+	Sata8 string
+	Sata9 string
+
+	Nets map[string]string
+	Net1 string
+	Net2 string
+	Net3 string
+	Net4 string
+	Net5 string
+	Net6 string
+	Net7 string
+	Net8 string
+	Net9 string
+
+	Unuseds map[string]string
+	Unused0 string
+	Unused1 string
+	Unused2 string
+	Unused3 string
+	Unused4 string
+	Unused5 string
+	Unused6 string
+	Unused7 string
+	Unused8 string
+	Unused9 string
+}
+
+func (s *VirtualMachineConfig) MergeFields() {
+	if nil == s.Ides || nil == s.Scsis || nil == s.Satas || nil == s.Nets || nil == s.Unuseds {
+		s.Ides = map[string]string{}
+		s.Scsis = map[string]string{}
+		s.Satas = map[string]string{}
+		s.Nets = map[string]string{}
+		s.Unuseds = map[string]string{}
+
+		t := reflect.TypeOf(*s)
+		v := reflect.ValueOf(*s)
+		count := v.NumField()
+
+		regIde, _ := regexp.Compile("^Ide[\\d]+$")
+		regScsi, _ := regexp.Compile("^Scsi[\\d]+$")
+		regSata, _ := regexp.Compile("^Sata[\\d]+$")
+		regNet, _ := regexp.Compile("^Net[\\d]+$")
+		regUnused, _ := regexp.Compile("^Unused[\\d]+$")
+
+		for i := 0; i < count; i++ {
+			fn := t.Field(i).Name
+			fv := v.Field(i).String()
+			//fmt.Println(fn, fv)
```

Comment:
either delete this or log it if you feel it has value

Reply:
used for find and delete all unuseds